docs/en/observability/monitor-k8s/monitor-k8s-application-performance.asciidoc

```diff
@@ -2,52 +2,220 @@
 [[monitor-kubernetes-application-performance]]
 == Part 3: Monitor application performance
 
-[Author: TBD]
+Quickly triage and troubleshoot application performance problems with the help of Elastic
+application performance monitoring (APM).
 
-TODO: Describe how to use APM to monitor applications.
+Think of a latency spike -- APM can help you narrow the scope of your investigation to a single service.
+Because you've also ingested and correlated logs and metrics, you can then link the problem to CPU and memory utilization via the error log entries of a particular Kubernetes pod.
 
 [discrete]
-=== Set up APM Server
+=== Step 1: Set up APM Server
 
-TODO: Describe how to set up APM server.
+Application monitoring data is streamed from your applications running in Kubernetes to APM Server,
+where it is validated, processed, and transformed into {es} documents.
 
-[discrete]
-==== Through ECK
+There are many ways to deploy APM Server when working with Kubernetes.
+This guide assumes you're using our hosted {ess} on {ecloud}.
+If you want to manage APM Server yourself, there are a few alternative options:
+
+* {eck-ref}/[Elastic Cloud on Kubernetes (ECK)] -- The Elastic recommended approach for managing
+APM Server deployed with Kubernetes.
+Built on the Kubernetes Operator pattern, ECK extends basic Kubernetes orchestration capabilities
+to support the setup and management of APM Server on Kubernetes.
+
+* Deploy APM Server as a DaemonSet -- Ensure a running instance of APM Server on each node in your cluster,
+like when a single APM Server instance should be reused for any pod that may be allocated to a node.
+
+* Deploy APM Server as a sidecar -- For environments that should not share an APM Server,
+like when directing traces from multiple applications to separate {es} clusters.
 
-Question: Are we sure we want to cover ECK here? Can we point to the ECK docs
-instead? If we try to document all the ways in all the sections, I think users
-might get confused.
+* {apm-server-ref-v}/installing.html[Download and install APM Server] -- The classic, non-Kubernetes option.
 
 [discrete]
-==== On cloud
+=== Step 2: Set up APM Agents
 
-TODO: Describe how to set up APM server on cloud.
+In most cases, setting up APM agents and thereby instrumenting your applications
+is as simple as installing a library and adding a few lines of code.
```

Comment:
general guideline is to avoid saying "simple" in docs in case someone is struggling with the task. :-)